arch/mips/boot/compressed/dbg.c

```diff
@@ -655,15 +655,15 @@ static void set_res(void)
 	smode1.prst = 0;
 
 	gs_write_smode1(smode1);
-	gs_writeq_smode2(0x0);
-	gs_writeq_srfsh(0x2);
-	gs_writeq_synch1(0x2c4d0160509d7);
-	gs_writeq_synch2(0x240300);
-	gs_writeq_syncv(0xb0e00002400004);
-	gs_writeq_display1(0x43777f000280f8);
+	gs_writeq_smode2(0x1);
+	gs_writeq_srfsh(0x8);
+	gs_writeq_synch1(0x7f5c21fc83030);
+	gs_writeq_synch2(0x3484bc);
+	gs_writeq_syncv(0xa9000502101401);
+	gs_writeq_display1(0x1c79ff0184a2a9);
```

Comment:
Hmm, I have `0x1f79`... but you have `0x1c79`... Where does your value come from? I compare with registers after `echo S:640x512i-50 >/sys/class/graphics/fb0/mode`.

Reply:
Ah, yes, maybe I dumped regs after gcont applies, maybe this is affecting display1

Reply:
Hmm... I think the gcont field is independent. Here’s what DISPLAY1 looks like for me:

```
# cat display1
# 0x1f79ff0184a2a9
dx 681
dy 74
magh 3
magv 0
dw 2559
dh 503
```

With custom margin adjustments, if you’ve applied such things, they’d be different compared to the built-in ps2fb preset mode.

Reply:
no, I didn't touch any other values. However, I applied video mode differently. I modified the init script:
`modprobe ps2fb mode_option=640x512i@50`
Maybe you see here some leftovers from applying 1920x1080p@60 ?

Reply:
Oh, haha, I know what it is. Press `ctrl` + `L` to clear the screen, and then try

```
for i in $(seq 100); do grep '^dh ' <display1; sleep 0.1; done
```

and notice how the `dh` field changes as the screen is scrolling. :-) That’s because the driver has

https://github.com/frno7/linux/blob/59a11ab94a4020408c7dfbf92bd55778e17b43f1/drivers/video/fbdev/ps2fb.c#L2149

which is hardware-accelerated vertical panning. When scrolling, no data is copied by the GS, it just adjusts the display offsets, which is _much_ more efficient. Since the driver uses both DISPLAY registers for scrolling as a virtual split-screen, it can do infinite vertical scrolling.

Reply:
oh, this sounds interesting :) but how to calculate proper values in that case?

Reply:
Easy, for our purpose I will, while reading the registers, temporarily remove

https://github.com/frno7/linux/blob/59a11ab94a4020408c7dfbf92bd55778e17b43f1/drivers/video/fbdev/ps2fb.c#L2148-L2151

from the `ps2fb` driver, so it stays fixed at the top of the screen (instead of scroll-wrapping the DISPLAY registers as it normally would). The hardware-accelerator flags are optional, fortunately.